android/pytorch_android/CMakeLists.txt

```diff
@@ -1,5 +1,172 @@
+#     ${pytorch_android_SOURCES}
+# )
+
+# if(APPLE)
+#   # Need to add rpath so dlopen can find dependencies.
+#   add_custom_command(TARGET pytorch_jni
+#       POST_BUILD COMMAND
+#       ${CMAKE_INSTALL_NAME_TOOL} -add_rpath "@loader_path"
+#         $<TARGET_FILE:pytorch_jni>)
+# endif()
+
+# target_compile_options(pytorch_jni PRIVATE
+#   -fexceptions
+# )
+
+# target_include_directories(pytorch_jni BEFORE
+#    PUBLIC $<BUILD_INTERFACE:${libtorch_include_DIR}>
+# )
+
+# set(fbjni_DIR ${CMAKE_CURRENT_LIST_DIR}/../libs/fbjni/)
+# set(fbjni_BUILD_DIR ${CMAKE_BINARY_DIR}/fbjni/${BUILD_SUBDIR})
+
+# add_subdirectory(${fbjni_DIR} ${fbjni_BUILD_DIR})
+
+# # ---[ Vulkan deps
+# if(USE_VULKAN)
+#   set(Vulkan_LIBS)
+#   set(Vulkan_INCLUDES)
+#   include(${CMAKE_CURRENT_LIST_DIR}/../../cmake/VulkanDependencies.cmake)
+# endif()
+
+# if(ANDROID_ABI)
+#   function(import_static_lib name)
+#     add_library(${name} STATIC IMPORTED)
+#     set_property(
+#         TARGET ${name}
+#         PROPERTY IMPORTED_LOCATION
+#         ${CMAKE_CURRENT_LIST_DIR}/src/main/jniLibs/${ANDROID_ABI}/${name}.a)
+#   endfunction(import_static_lib)
+
+#   import_static_lib(libtorch)
+#   import_static_lib(libtorch_cpu)
+#   import_static_lib(libc10)
+#   import_static_lib(libnnpack)
+#   import_static_lib(libXNNPACK)
+#   import_static_lib(libpytorch_qnnpack)
+#   import_static_lib(libpthreadpool)
+#   import_static_lib(libeigen_blas)
+#   import_static_lib(libcpuinfo)
+#   import_static_lib(libclog)
+
+#   # Link most things statically on Android.
+#   set(pytorch_jni_LIBS
+#       fbjni
+#       -Wl,--gc-sections
+#       -Wl,--whole-archive
+#       libtorch
+#       libtorch_cpu
+#       -Wl,--no-whole-archive
+#       libc10
+#       libnnpack
+#       libXNNPACK
+#       libpytorch_qnnpack
+#       libpthreadpool
+#       libeigen_blas
+#       libcpuinfo
+#       libclog
+#   )
+# else()
+#   # Prefer dynamic linking on the host
+#   set(pytorch_jni_LIBS
+#       fbjni
+#       torch
+#       torch_cpu
+#       c10
+#       cpuinfo
+#       clog
+#   )
+
+#   if(USE_NNPACK)
+#     list(APPEND pytorch_jni_LIBS nnpack)
+#   endif()
+
+#   if(USE_XNNPACK)
+#     list(APPEND pytorch_jni_LIBS XNNPACK)
+#   endif()
+
+#   if(USE_SYSTEM_PTHREADPOOL)
+#     list(APPEND pytorch_jni_LIBS pthreadpool)
+#   endif()
+
+#   if(USE_PYTORCH_QNNPACK)
+#     list(APPEND pytorch_jni_LIBS pytorch_qnnpack)
+#   endif()
+
+# endif()
+
+# if(USE_VULKAN)
+#   list(APPEND pytorch_jni_LIBS ${Vulkan_LIBS})
+# endif()
+
+# target_link_libraries(pytorch_jni ${pytorch_jni_LIBS})
+
+# install(TARGETS pytorch_jni
+#   ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+#   LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+#   RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}) #For windows
+
+# if(MSVC)
+#   install(FILES $<TARGET_PDB_FILE:pytorch_jni> DESTINATION ${CMAKE_INSTALL_LIBDIR} OPTIONAL)
+#   install(TARGETS pytorch_jni DESTINATION ${CMAKE_INSTALL_LIBDIR})
+# endif()
+
+project(pytorch_jni_lite CXX)
```

Comment:
Make it optional and uncomment the lines above?

Reply:
Yes I should do that..

Reply:
Update it by adding a switch to use the environment variable. It's more complicated than a switch because the environment variable need to go through -> build.gradle -> `android/pytorch_android/CMakeLists.txt`